optuna/pruners/hyperband.py

```diff
@@ -93,25 +98,23 @@ class HyperbandPruner(BasePruner):
     def __init__(
         self,
         min_resource: int = 1,
-        max_resource: int = 80,
+        max_resource: Union[str, int] = "auto",
         reduction_factor: int = 3,
         n_brackets: Optional[int] = None,
         min_early_stopping_rate_low: Optional[int] = None,
     ) -> None:
 
-        self._pruners = []  # type: List[SuccessiveHalvingPruner]
+        self._min_resource = min_resource
+        self._max_resource = max_resource
         self._reduction_factor = reduction_factor
+        self._n_brackets = n_brackets
+        self._min_early_stopping_rate_low = min_early_stopping_rate_low
+        self._pruners = []  # type: List[SuccessiveHalvingPruner]
         self._total_trial_allocation_budget = 0
+        self._trial_allocation_budgets = []  # type: List[int]
+        self._running_trials = {}  # type: Dict[int, int]
```

Comment:
Where do you use this?

Reply:
Thank you for your comment. The `self._trial_allocation_budgets` is used in `self._try_initialization()` and `self._get_bracket_id()`. The `self._running_trials` is never used. I will remove this.